Add quantity negative

afh opened the pull request:
This PR builds on top of #21 and includes those changes, until #21 has been reviewed and ideally merged, this is a draft PR.

This PR adds the `quantity_negative` capture to allow for custom highlighting of negative quantities. Personally I find it very helpful to see positive quantities in black and negative ones on bold red.

afh committed:
Add quantity negative capture

afh commented on queries/highlights.scm:
```diff
@@ -1,16 +1,13 @@
+((date) @number.date)
+((interval) @number.interval)
+((quantity) @number.quantity)
+((quantity_negative) @number.negative.quantity)
```
Apart from this line above ⬆️ the changes in this file are proposed in #21

cbarrete replied:
I think that `@number.quantity.negative` makes more semantic sense

cbarrete commented on grammar.js:
```diff
@@ -254,17 +254,21 @@ module.exports = grammar({
 
         quantity: $ => seq(
-            optional('-'), /\d([\d., ]*\d)?/,
+            optional('+'), /\d([\d., ]*\d)?/,
```
Can you extract `/\d([\d., ]*\d)?/` as `unsigned_quantity` or something? I'd like to avoid the proliferation of nasty regexes where possible :)

Also, while you're at it, do you mind renaming `quantity_negative` to `negative_quantity`?

afh replied:
Excellent suggestions, thanks @cbarrete. This PR has been updated accordingly. I decided to name the extracted regex `_quantity`, but am happy to change it if you believe `unsigned_quantity` is a more suitable name.

cbarrete replied:
That's fine, thanks :)

afh committed:
Rename quantity_negative to negative_quantity

Extract quantity regex into _quantity to avoid proliferation of
regexes where possible.
